checkers/hugeParam_checker.go

```diff
@@ -49,6 +50,9 @@ func (c *hugeParamChecker) checkParams(params []*ast.Field) {
 	for _, p := range params {
 		for _, id := range p.Names {
 			typ := c.ctx.TypeOf(id)
+			if _, ok := typ.(*types.TypeParam); ok {
```

Comment:
Shouldn't we use a polyfill like `x/exp/typeparams` to make this code compileable with older Go versions?
golangci-lint can't deprecate pre-generics release yet AFAIK.

Reply:
https://pkg.go.dev/golang.org/x/exp/typeparams

Reply:
Fixed.

If we support go 1.17 then I should hide the new test cases using build constraints.

Reply:
BTW, If you'd like I can add go1.18 CI support in a separated PR.

https://github.com/mcdoker18/go-critic/pull/3/commits/2041b982633b256b73c61521614172b11133f1dd